objects/src/transaction/prepared_tx.rs

```diff
@@ -25,8 +29,9 @@ impl PreparedTransaction {
         program: Program,
         tx_script: Option<TransactionScript>,
         tx_inputs: TransactionInputs,
+        note_args: Option<BTreeMap<NoteId, Word>>,
```

Comment:
Why do we need to add `note_args` here? I think we can remove them.

Reply:
But doesn't it make more sense to have it as part of the prepared transaction? Note args seem to be part of the transaction

Reply:
That's a good point. I think note args basically fall into the same category as `tx_script` - i.e., they are a part of the transaction data that needs to be known only the the prover (the verifier doesn't need that).

I think ideally we want to consolidate these into a separate struct - e.g., something like `TransactionArgs` (or maybe some other name). This could look as follows:
```Rust
pub struct TransactionArgs {
    /// This would be our current TransactionScript but with `inputs` field removed
    tx_script: Option<TransactionScript>,
    note_args: BTreeMap<NoteId, Word>,
    /// this would contain additional data that needs to go into the advice provider, including
    /// transaction script inputs, but also could contain pre-images of note args, if needed
    advice_inputs: AdviceInputs,
}
```
We don't need to implement all of this in this PR, but I wonder if it makes to do a simplified version of this to avoid changing the same code twice. The simplified version could look like:
```Rust
pub struct TransactionArgs {
    tx_script: Option<TransactionScript>, // our current TransactionScript with no changes
    note_args: BTreeMap<NoteId, Word>,
}
```
But either way, I think you are right and we should make `note_args` a part of `PreparedTransaction`.

Reply:
Ok, but why would we not have `note_args` as an option, like in 

```Rust
pub struct TransactionArgs {
    tx_script: Option<TransactionScript>, // our current TransactionScript with no changes
    note_args: Option<BTreeMap<NoteId, Word>>,
}
```

it should be applied for the same reasons as for `tx_script`.

Reply:
Ah yes, correct. It should be:
```Rust
pub struct TransactionArgs {
    tx_script: Option<TransactionScript>,
    note_args: BTreeMap<NoteId, Word>,
}
```

Reply:
Ok, I added
```Rust
  pub struct TransactionArgs {
      tx_script: Option<TransactionScript>,
      note_args: BTreeMap<NoteId, Word>,
  }
```

I needed to add them to the `PreparedTransaction` and the `ExecutedTransaction` as the `ExecutedTransaction` must `... contain all the information required to re-execute`.